### PR TITLE
Re-org the codes about blocks

### PR DESCRIPTION
**Pull request by EanCuznaivy**
Tuned some properties

### CODE REVIEW

File: AElf.Kernel/Block.cs [hunk 1]
```diff
@@ -19,20 +20,34 @@ public class Block : IBlock
...
         {
             BlockHeader = new BlockHeader(preBlockHash);
         }
 
+        /// <summary>
+        /// Add a transaction to block body.
+        /// </summary>
+        /// <param name="tx"></param>
+        /// <returns></returns>
         public bool AddTransaction(ITransaction tx)
```
Comment (loning) [inline AElf.Kernel/Block.cs]: we need to add tx id to block header merkle tree

File: AElf.Kernel/BlockHeader.cs [hunk 2]
```diff
@@ -1,32 +1,26 @@
...
-                return GetTransactionMerkleTreeRoot();
-            }
-        }
-
-        private MerkleTree<ITransaction> MerkleTree { get; set; } = new MerkleTree<ITransaction>();
-
+        /// <summary>
+        /// Time stamp.
+        /// </summary>
         public long TimeStamp => DateTime.UtcNow.Second;
```
Comment (loning) [inline AElf.Kernel/BlockHeader.cs]: DateTime.UtcNow.Second is not right

File: AElf.Kernel/BlockHeader.cs [hunk 3]
```diff
@@ -42,16 +36,32 @@ public BlockHeader(IHash<IBlock> preBlockHash)
         /// </summary>
         public int Nonce { get; set; }
 
+        #region Private fields
+        private MerkleTree<ITransaction> TransactionTrie = new MerkleTree<ITransaction>();
```
Comment (loning) [inline AElf.Kernel/BlockHeader.cs]: private filed should be _transactionTrie, or _merkleTreeOfTransactions

File: AElf.Kernel/BlockHeader.cs [hunk 4]
```diff
@@ -62,8 +72,12 @@ public IHash<IBlockHeader> GetHash()
             return new Hash<IBlockHeader>(this.GetSHA256Hash());
         }
 
-        public void SetNonce()
+        /// <summary>
+        /// Adjust the value of Nonce while mining.
+        /// </summary>
+        public void AdjustNonceWhileMining()
```
Comment (loning) [inline AElf.Kernel/BlockHeader.cs]: do not add any mining method in the implement of Block

File: AElf.Kernel/BlockHeader.cs [hunk 5]
```diff
@@ -62,8 +72,12 @@ public IHash<IBlockHeader> GetHash()
             return new Hash<IBlockHeader>(this.GetSHA256Hash());
```
Comment (loning) [inline AElf.Kernel/BlockHeader.cs]: Should just this.GetHash(), not GetSHA256Hash

File: AElf.Kernel/ITransactionReceiver.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AElf.Kernel
+{
+    public interface ITransactionReceiver
```
Comment (loning) [inline AElf.Kernel/ITransactionReceiver.cs]: we do not need ITransactionReceiver in Kernel

File: AElf.Kernel/ITransactionSender.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿namespace AElf.Kernel
+{
+    public interface ITransactionSender
```
Comment (loning) [inline AElf.Kernel/ITransactionSender.cs]: we do not need ITransactionSender

File: AElf.Kernel/LiteDBDataProvider.cs [hunk 8]
```diff
@@ -23,29 +23,29 @@ public byte[] Serialize()
...
             task.Start();
             return await task;
         }
 
         async Task IAccountDataProvider.SetAsync(IHash key, ISerializable obj)
         {
-            var c = this.db.GetCollection<Record>("data");
-            Task<bool> task = new Task<bool>(() => c.Upsert(new Record(){Key = key, Value = obj.Serialize()}));
+            var c = db.GetCollection<Record>("data");
+            Task<bool> task = new Task<bool>(() => c.Upsert(new Record() { Key = key, Value = obj.Serialize() }));
```
Comment (loning) [inline AElf.Kernel/LiteDBDataProvider.cs]: use Task.Factory to start new task

File: AElf.Kernel/Miner.cs [hunk 9]
```diff
@@ -7,8 +7,8 @@ public byte[] Mine(IBlockHeader blockheader)
             int bits = (blockheader as BlockHeader).Bits;
             while (true)
             {
-                //Change the nonce
-                (blockheader as BlockHeader).SetNonce();
+                //Change the Nonce
+                (blockheader as BlockHeader).AdjustNonceWhileMining();
```
Comment (loning) [inline AElf.Kernel/Miner.cs]: AdjustNonceWhileMining should be implement in some class implemented IMiner

File: AElf.Kernel/BlockHeader.cs (hunk 3 above)
Comment (samfrb) [reply AElf.Kernel/BlockHeader.cs]: This is purely coding convention, but I agree with it so I will put it in the style guide.

**Pull request closed without merging**